### PR TITLE
Makes it so Mechas can't pass through plastic flaps.

### LINKED ISSUE #10741: Mechs and plastic flaps
I've just seen a mech walk through plastic flaps
Wtf.
Humans can't fit
Why should a huge mech fit?

### PR DESCRIPTION
**Pull request by Firecage**
This PR makes it so that Mechas can no longer pass through the Plastic Flaps. Everyday the Plastic Flaps grows stronger.

Fixes #10741

:cl: Firecage
tweak: Mechas can no longer pass through Plastic Flaps
/:cl:

### CODE REVIEW

File: code/game/objects/structures/plasticflaps.dm [hunk 1]
```diff
@@ -23,14 +23,20 @@
...
 		return 0
 
-	if (istype(A, /obj/structure/closet/cardboard))
+	if(istype(A, /obj/structure/closet/cardboard))
 		var/obj/structure/closet/cardboard/C = A
 		if(C.move_delay)
 			return 0
 
+	if(istype(A, /obj/mecha))
+		if(istype(A,/obj/mecha/combat/phazon)) //Since it is a bluespace mech, silly flaps can't stop it.
```
Comment (GunHog) [inline code/game/objects/structures/plasticflaps.dm]: Not needed. Just put the mech in Phaze mode and it does not care about **any barriers**.

Comment (Firecage) [inline code/game/objects/structures/plasticflaps.dm]: I know. Just thought it would be a fun inclusion to make them pass through flaps by default.

Comment (KorPhaeron) [inline code/game/objects/structures/plasticflaps.dm]: Bad snowflake code

Comment (Incoming5643) [inline code/game/objects/structures/plasticflaps.dm]: Everything about plastic flaps is snowflake.

Comment (Firecage) [inline code/game/objects/structures/plasticflaps.dm]: @KorPhaeron All of Plastic Flap code is snowflake code though. And I am definitely not rewriting this mess, so this is a fix using what is already there.

**Comment by Dorsisdwarf** [comment]
grrrr

**Comment by AdamElTablawy** [comment]
at some point plastic flaps will just become resprited walls that can't be deconstructed

**Comment by Firecage** [comment]
.....To be fair....thats kinda what they are, bar that some things can pass through.

**Comment by Supermichael777** [comment]
REBALENCE DETECTED WEWOWEWO

**Comment by Armhulen** [comment]
flap powercreep :+1:

**Comment by Kromgar** [comment]
Why wouldn't mechs be able to push aside plastic flaps?

**Comment by kilkun** [comment]
most mechs that get built can punch a hole through walls or at the very least drill through them

:-1: useless pr

**Comment by jughu** [comment]
Not every mech can punch a hole in the wall and the drill is very limited to normal walls and airlocks (if you like to go trough the front door) reinforced walls they cant drill until diamond
Of course you can get into everything with high science rendering flaps even more iseless since you can just jackhammer your way trough everywhere

**Comment by kilkun** [comment]
re-read what i wrote.
MOST mechs that GET BUILT
barely anybody wastes time and resources making odysseuses and literally NOBODY makes firefighters because they are ripleys who accept less equipment.

Durands and decked out ripleys are the hot shit.

also why the fuck are medical mechs called odysseus? what a terrible fucking name.

**Comment by optimumtact** [comment]
So punch through the flap.

**Comment by Chocobro** [comment]
odysseus is a better name than kilkin imo
greek hero in comparison to random nerdo on the internet

**Comment by kilkun** [comment]
it makes zero sense for the medical mech to be named it tbh

**Comment by Chocobro** [comment]
oh well?

**Comment by kilkun** [comment]
if you didnt have a reason why did you reply

**Comment by Chocobro** [comment]
why didn't you already make an angry ree-tier pr yelling how the odysseus should be called xXmercyxkillax420xoxo
because we're losers on the internet
that's why

**Comment by kilkun** [comment]
because for some reason i highly doubt it'd get merged

**Pull request merged**